File: front-end/src/app/rc/rack/rack.service.js
(function() {

    angular.module('app.rc.rack').service('rackService', rackService);

    function rackService(rackFactory) {

        return {
            getRacks: getRacks,
            getRack: getRack,
            saveRack: saveRack,
            deleteRack: deleteRack,
        }

        function getRacks() {
            return rackFactory.list({});
        }

        function getRack(id) {
            return rackFactory.get({id},null);
        }

        function saveRack(rack) {
            return rackFactory.save({},rack);
        }

        function deleteRack(id) {
            return rackFactory.delete({id},null);
        }

    }

})();

File: api/src/main/java/com/api/service/RackService.java
package com.api.service;

import com.api.dto.RackDTO;
import com.api.model.Rack;

import java.util.List;

public interface RackService {

    Rack dtoToEntity(RackDTO rackDTO);

    List<RackDTO> findAll();

    Rack get(Long id);

    Rack save(RackDTO rack);

    //Rack update(RackDTO rack);

    void delete(Long id);
}
